classic_cp/kidneyexchange/kidneyexchange.jl

```diff
@@ -0,0 +1,278 @@
+using SeaPearl
+
+"""
+Instances format
+4 0.33 #numberOfPairs density
+2 4    #pair 1 can receive a kidney from pairs 2 and 4
+1      #pair 2 can receive a kidney from pair 1
+4      #pair 3 can receive a kidney from pair 4
+3      #pair 4 can receive a kidney from pair 3
+"""
+
+struct InputData
+    numberOfPairs         :: Int
+    density               :: Float16
+    compatibilities       :: Vector{Union{Vector{Int}, Nothing}}
+end
+
+include("IOmanager.jl")
+
+"""
+    Unfinished version, problem defining the objective (how to convert IntVar to Bool?)
+"""
+function solve_kidneyexchange_bis(filename::String)
+
+    InputData = getInputData(filename)
+
+    n = InputData.numberOfPairs
+    c = InputData.compatibilities
+
+    trailer = SeaPearl.Trailer()
+    model = SeaPearl.CPModel(trailer)
+
+    ### Try to reduce problem ###    
+    canReduce = true
+    current_n = n
+    while canReduce
+        flatten_c = reduce(vcat,c)
+        usefullPairs = [i for i in 1:n if i in flatten_c && c[i] != []] 
+        if length(usefullPairs) < current_n
+            c = [filter(e -> e in usefullPairs, list) for list in c]
+            current_n = length(usefullPairs)
+        else
+            canReduce = false
+        end
+    end
+
+    count = 1
+    pairsEquivalence = []
+    for i in 1:n
+        if isempty(c[i])
+            push!(pairsEquivalence, nothing)
+        else
+            push!(pairsEquivalence, count)
+            count += 1
+        end
+    end
+
+    c = filter(e -> !isempty(e), c)
```

Comment:
I guess this part was meant to remove all Pairs that could neither give nor receive in any case ?

Reply:
Exactly, I will add some comments to clarify the instance reduction

Reply:
done